File: leetcode2020/0014.最长公共前缀/0014-最长公共前缀.py
class Solution:
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if strs==[]:
            return ''
        prefix=strs[0]
        for i in range(1,len(strs)):
            if not prefix:
               return ''
            else:
               while prefix not in strs[i][:len(prefix)] and len(prefix)>0:
                   prefix=prefix[:len(prefix)-1]
        return prefix

File: leetcode2020/0014.最长公共前缀/0014-最长公共前缀.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty()) return string();
        sort(strs.begin(), strs.end());
        string start = strs.front(), end = strs.back();
        int i,num = min(start.size(), end.size());
        for(i = 0; i < num && start[i] == end[i]; i++);
        return start.substr(0,i);
    }
};